calendar-scheduling/src/main/java/com/linagora/calendar/scheduling/AlarmEventScheduler.java

```diff
@@ -0,0 +1,124 @@
+ *  useful, but WITHOUT ANY WARRANTY; without even the implied      *
+ *  warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR         *
+ *  PURPOSE. See the GNU Affero General Public License for          *
+ *  more details.                                                   *
+ ********************************************************************/
+
+package com.linagora.calendar.scheduling;
+
+import java.io.Closeable;
+import java.time.Clock;
+import java.time.Duration;
+import java.util.concurrent.ThreadLocalRandom;
+
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+
+import org.apache.james.lifecycle.api.Startable;
+import org.apache.james.util.ReactorUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linagora.calendar.scheduling.AlarmEventSchedulerConfiguration.Mode;
+import com.linagora.calendar.storage.AlarmEvent;
+import com.linagora.calendar.storage.AlarmEventDAO;
+import com.linagora.calendar.storage.AlarmEventLocker;
+
+import reactor.core.Disposable;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class AlarmEventScheduler implements Startable, Closeable {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AlarmEventScheduler.class);
+
+    private final Clock clock;
+    private final AlarmEventDAO alarmEventDAO;
+    private final AlarmEventLocker alarmEventLocker;
+    private final AlarmTriggerService alarmTriggerService;
+    private final AlarmEventSchedulerConfiguration configuration;
+
+    private Disposable loop;
+
+    @Inject
+    @Singleton
+    public AlarmEventScheduler(Clock clock, AlarmEventDAO alarmEventDAO,
+                               AlarmEventLocker alarmEventLockerCandidate,
+                               AlarmTriggerService alarmTriggerService,
+                               AlarmEventSchedulerConfiguration configuration) {
+        this.clock = clock;
+        this.alarmEventDAO = alarmEventDAO;
+        this.alarmTriggerService = alarmTriggerService;
+        this.configuration = configuration;
+
+        if (Mode.SINGLE.equals(configuration.mode())) {
+            alarmEventLocker = AlarmEventLocker.NOOP;
+        } else {
+            alarmEventLocker = alarmEventLockerCandidate;
+        }
+    }
+
+    public void start() {
+        Duration initialDelay = Duration.ofMillis(ThreadLocalRandom.current().nextLong(0,
+            configuration.initialJitterMax().toMillis()));
+
+        LOGGER.info("Starting AlarmEventScheduler: initialDelay={}, pollInterval={}, batchSize={}",
+            initialDelay, configuration.pollInterval(), configuration.batchSize());
+
+        loop = Flux.interval(initialDelay, configuration.pollInterval())
+            .onBackpressureDrop()
+            .concatMap(tick -> pollAndProcess()
+                .onErrorResume(ex -> {
+                    LOGGER.warn("pollAndProcess failed: {}", ex.toString(), ex);
```

Comment:
```suggestion
                    LOGGER.warn("pollAndProcess failed", ex);
```

We do not need that duplicated message in the logs